vllm/inputs.py

```diff
@@ -0,0 +1,108 @@
+from typing import (TYPE_CHECKING, List, Literal, Optional, Sequence,
+                    TypedDict, Union, cast, overload)
+
+if TYPE_CHECKING:
+    from vllm.sequence import MultiModalData
+
+
+class ParsedString(TypedDict):
+    text: str
+    is_tokens: Literal[False]
+
+
+class ParsedTokens(TypedDict):
+    text: List[int]
+    is_tokens: Literal[True]
```

Comment:
Why do we need to make these two classes? Can we not have one class that's simply

```
class ParsedPrompt(TypedDict):
    data: Union[List[int], str]
    is_tokens: bool
```

Reply:
This is mainly to help the type checker since it's not able to perform type narrowing between `List[int]` and `str` without the use of tagged unions.